### ISSUE TITLE
Return to current folder for file browser

### ISSUE THREAD

**Issue by roed314** [sagemathinc/smc#423]
When I use the file browser to open a file that's not in my home directory, go edit the file, then return to the file browser, it takes me back to my home directory.  This behavior is annoying when I want to open another file in the same folder, which is nested many folders deep (e.g. sage/src/sage/rings/finite_rings/constructor.py and sage/src/sage/rings/finite_rings/finite_field_base.pyx).

It's pretty easy to get back to the home folder: just click on the home icon.  I would suggest having the file browser remain viewing the same folder if you click away and then return.

**Comment by haraldschilly** [comment]
Hi @roed314 ... this is something that's also on my wishlist. I just can't find the ticket.

However, the story behind this is that is mimics the behavior of emacs and in some sense it helps you navigating around. From an UX standpoint, it's a bit wrong because it does something automatically without telling the user, and that's when the "internal model" in your brain of the state diverges from the actual state without a hint → confusion.

It's also not working all the time, e.g. when you open a file directly, you are still in $HOME while when you go back and forth between tabs, only then the directory change occurs. Whoever works on this should also fix this missing action hook.

Long story short: I'm assigning this to myself. The implementation path would be to add a boolean user setting and then only do this if it is set. Only question remains what the default should be. I vote for `False`.

**Comment by roed314** [comment]
Cool.  I would also vote for the behavior I'm requesting to be the default.  :-)

**Comment by williamstein** [comment]
I also vote for False.  I like the current behavior, but it shouldn't be
the default.  (Emacs isn't the dominant operating system, after all!)

On Wed, Feb 17, 2016 at 8:42 AM, roed314 <notifications@github.com> wrote:

> Cool. I would also vote for the behavior I'm requesting to be the default.
> :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sagemathinc/smc/issues/423#issuecomment-185291106>.
>



-- 
William (http://wstein.org)

**Comment by williamstein** [comment]
The fix mentioned above at least makes the current behavior (the emacs style behavior) consistent.
